### ISSUE TITLE
New Turnigy accucell 6 80W

### ISSUE THREAD

**Issue by aequitas** [stawel/cheali-charger#106]
I just bought the new accucell 80W (with XT60 input connector and usb port), http://www.hobbyking.com/hobbyking/store/__64345__Turnigy_Accucel_6_80W_10A_Balancer_Charger_LiHV_Capable.html

After successfully running cheali on a clone iMax B6 I am so fond of it I want it on the turnigy as well.

Does anyone have experience with running cheali on this platform? As far as I could see there is no ISP interface. There is room for a 5 pin header at the top, it could be an ISP (haven't traced the lines yet), but is missing one pin. It does seem to have Vcc and Gnd.

So far i had no luck connecting to the USB port, since the device won't be recognized under OSX:

    kernel[0]: USBF:    62715.168    The IOUSBFamily is having trouble enumerating a USB device that has been plugged in.  It will keep retrying.  (Port 1 of Hub at 0x14000000)
    kernel[0]: USBF:    62715.168    The IOUSBFamily was not able to enumerate a device.

![_mg_6460](https://cloud.githubusercontent.com/assets/365827/7898033/a93c34aa-06f3-11e5-8250-a4b8d9421215.jpg)

**Comment by aequitas** [comment]
When starting the device in update mode (hold '-' and 'enter' during start) it does seem to register with the os as a 'Megawin Technologies' MA108 HID device (http://www.megawin.com.tw/zh-tw/product/productDetail/MA108), product id: 0x030b, vendor id: 0x0e6a.

**Comment by stawel** [comment]
If a megawin CPU is the main processor we probably will not support your charger,
the CPUs architecture is just too old, but it may only be used for USB communication,
in this case you should check what is under the LCD.

Best Regards,
Paweł

**Comment by aequitas** [comment]
@stawel i will see if i can get the lcd off to see whats under there.

**Comment by aequitas** [comment]
It's a megawin under there, MA84G564AD4, so no luck :(

![_mg_6506](https://cloud.githubusercontent.com/assets/365827/8149801/8f8d3e70-12d1-11e5-8fd7-79e1ddb1c7a6.jpg)

**Comment by schavkin** [comment]
Hobbyking accucell 6 50W LiHV has same MPU as 80W version.

**Comment by stawel** [comment]
please let me know, 
if someone of you would like to add this MPU to the project,
(unfortunately  this will cost you a lot of free time)

**Comment by NIXIE-123** [comment]
Hi!
I have accucel 6 50W 6A lihv. And have the same problem. There is no option to calibrate. That's pitty, because balancer is inaccurate. I've read, that is possible to change firmware via usb. I'm waiting if (hopefully) someone will work on this, because those chargers are getting popular.
Best regards,
Pozdrawiam

**Comment by Mikimimi** [comment]
Hi,
i have Imax b6 AC V2 that also has megawin MA84G564. It is genuine SkyRC. The charger is off by 0,4V. I would like cheali on this charger too.

Here is the picture of the board [https://lh6.googleusercontent.com/proxy/QrsrvHqYg-u1As7IhC4pMT4c-nvTGiDhdVjL-qzNXh1YDWbDJpY7Qd9zsuP4kazvK4sp0C8hTwRV07Q_fFHD1d0XvhE4r-zSOyFXOePq=w5000-h5000](url)
Here is picture of pinout [https://lh6.googleusercontent.com/proxy/9RIRwpSYj6fNPvq7Gvg4qfi48pbyDKaOOeQoC_Lehb-YNSPRpu9Y_7VDC2BnqgIdUD598tU08sZeI0Yw1lomJZR39y1Re5ehJKZkOlP6AFJtbEo=w5000-h5000](url)

Here are videos:
[https://www.youtube.com/watch?v=cPoWtJsmpwo](url)
[https://www.youtube.com/watch?v=ZhkbdxS_noY](url)
[https://www.youtube.com/watch?v=FJisTvgV2mk](url)

**Comment by stawel** [comment]
Hi guys,
I have bad news for you,
I've decided to drop the attempts to add support for the mcs-51 CPUs,
reason: currently the cheali-charger architecture is 
to complex for such a simple CPU, also the open source C compile ssdc 
has very poor optimization capabilities.

All my test code is stored in this branch:
https://github.com/stawel/cheali-charger/tree/mcs-51  
I managed to:
1. implement a simple  C++ to C translator (ssdc support only C)
2. rewrite the code so it can be complied with ssdc 
3. reduce the RAM usage to ~750B
I also:
4. disabled all possible features (including calibration)
5. added an empty mcs-51 hardware implementation (all hardware specific methods are empty)

After all of that  I was able to compile the code but  the code size was about 35kB.
Looking at the disassembled code it is clear that the ssdc compiler
is not suitable for such a big project - unfortunately we grew too much
(it's probably not the compilers fault, the CPU's architecture is just too simple)

nevertheless It looks like it should be possible to run cheali-charger on MA84G564,
but I hoped to go down with the code size below 32kB so it could be run also
on other mcs-51 CPUs.
Given that I had to almost entirely rewrite the cheali-charger code (point 1. and 2.) and the only 
benefit would be the support for one charger, I've decided to drop these efforts.

Sorry guys,

Best Regards,
Paweł

**Comment by aequitas** [comment]
Hi Paweł,

thank you very much for the effort you put into this! To bad it doesn't work out for us, but it's understandable. 

grtz,

Johan

**Comment by LTMNO** [comment]
![img_1625](https://user-images.githubusercontent.com/4542894/45091873-becc7280-b0e1-11e8-95a9-f6b80dc8c915.jpg)

Dooh... i have an old original.... ;-(
Is the problem the architecture of the code(upgrade) with this chip.. or the updating process?

**Comment by Baterka** [comment]
Accucel 6 80W HW version 1.00 still not supported? I really hate the min. 3.0V cutoff voltage for LiPo DISCHARGE... Many today's cells are 2.5V

**Comment by diggit** [comment]
If it has same mcu as in photo above, then it's not supported.

**Comment by diggit** [comment]
Typical Lixx cells have minimal amount of energy below 3.0V. It's not worth to discharge cells so deeply.

**Comment by Baterka** [comment]
> Typical Lixx cells have minimal amount of energy below 3.0V. It's not worth to discharge cells so deeply.

Not really agree. Today's 18650 are rated to 2.5V safe discharge...

**Comment by diggit** [comment]
Yes they have 2.5V cutoff. I am just saying, it's not worth to discharge cell so deeply. Eg. Sony VTC5A (2.0 V cutoff) has less than 4% SOC at 3V and close to 0% SOC at 2.8V.
There are probably cells with different characteristics.
But we are getting OT here.
